### PR TITLE
Making ball a static obstacle

### PR DESCRIPTION
**Pull request by rishiso**
## Description
The STOP play state is currently not working because the robots are still able to go within the zone of the ball. We resolved this issue by making the ball a static obstacle instead of a dynamic obstacle.

## Associated / Resolved Issue
Resolves [ClickUp card](https://app.clickup.com/t/86ayj3tzd)

## Steps to Test
Run the simulator in the STOP state. The robots should not be in the zone of the ball.

## Review Checklist

- [x] **Docstrings**: All methods and classes should have the file appropriate docstrings which follow the guidelines in the ["Contributing" page](https://rj-rc-software.readthedocs.io/en/latest/contributing.html) of our docs.
- [x] **Remove extra print statements**: Any print statements used for debugging should be removed
- [x] **Tag reviewers**: Tag some people for review and ping them on Slack

### CODE REVIEW

**Review by sid-parikh — CHANGES_REQUESTED** [review]
Can you update this to inflate the ball radius based on ball velocity the way we do for `their_robots`?

**Comment by rishiso** [comment]
Don't merge this yet. Just realized a potential issue

File: soccer/src/soccer/planning/planner/plan_request.cpp [hunk 1]
```diff
@@ -65,14 +65,21 @@ void fill_obstacles(const PlanRequest& in, rj_geometry::ShapeSet* out_static,
         out_static->add(std::make_shared<rj_geometry::Circle>(obs_center, obs_radius));
     }
 
+    // Adding ball as a static obstacle (because dynamic obstacles are not working)
+    // Only added when STOP state is enabled
+    double ball_vel_scaled = in.world_state->ball.velocity.mag() * .1;
+    double radius = kBallRadius + in.min_dist_from_ball * (1.0 + ball_vel_scaled);
```
Comment (sid-parikh) [inline soccer/src/soccer/planning/planner/plan_request.cpp]: can you leave a comment explaining this math? are you sure it's equivalent to the math used in fill_robot_obstacle? 

I know we add a min_dist, but we should start with that min_dist and then inflate+offset using the same math as fill_robot_obstacle (you can honestly just call that function), this ensures we stay min_dist away from a moving target by adding extra buffer proportional to velocity.

**Review by sid-parikh — CHANGES_REQUESTED** [review]

File: soccer/src/soccer/planning/planner/plan_request.cpp (hunk 1 above)
Comment (rishiso) [reply soccer/src/soccer/planning/planner/plan_request.cpp]: The math I was using previously was an equivalent version. However, I changed it to use the same math as the other function.

**Review by sid-parikh — APPROVED** [review]

**Comment by sid-parikh** [comment]
I approved PR to waive my "requested changes". However, since I authored a major portion of it, please do not merge until somebody else has reviewed.

**Review by p-nayak11 — APPROVED** [review]
Fix the GitHub Actions, but other than that LGTM

**Pull request merged**